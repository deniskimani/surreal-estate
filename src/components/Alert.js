import "../styles/alert.css";

const Alert = ({ message, success }) => {
  return (
    <div className={`alert alert-${success ? "success" : "error"}`}>
      {message}
    </div>
  );
};

export default Alert;
