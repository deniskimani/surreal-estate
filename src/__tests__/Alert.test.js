import { render } from "@testing-library/react";
import Alert from "../components/Alert";

describe("test alert component", () => {
  test("displays an error message", () => {
    const { getByText } = render(<Alert message="Error!" success={false} />);

    expect(getByText(/Error/).textContent).toBe("Error!");
  });
  test("displays a success message", () => {
    const { getByText } = render(<Alert message="Success!!!!" />);

    expect(getByText(/Success/).textContent).toBe("Success!!!!");
  });
  test("does not render an error or a success message if message props is empty", () => {
    const { asFragment } = render(<Alert message="" />);
    const alert = asFragment();
    expect(alert).toMatchSnapshot();
  });

  test("test alert", () => {
    const alert = {
      message: "",
      isSuccess: false,
    };
    const { asFragment } = render(
      <Alert message={alert.message} success={alert.isSuccess} />
    );

    expect(asFragment).toMatchSnapshot();
  });
});
