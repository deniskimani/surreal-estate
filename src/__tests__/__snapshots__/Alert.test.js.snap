// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test alert component does not render an error or a success message if message props is empty 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  />
</DocumentFragment>
`;

exports[`test alert component test alert 1`] = `[Function]`;
